airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/test/util/IntegrationTest.kt

```diff
@@ -54,6 +56,22 @@ abstract class IntegrationTest(
     // stream name doesn't need to be randomized, only the namespace.
     val randomizedNamespace = "test$timestampString$randomSuffix"
 
+    // junit is a bit wonky with injecting TestInfo.
+    // You can declare it as a constructor param, but you get a TestInfo instance
+    // that doesn't know what test method it's running in.
+    // You can also declare it as a param on the test function, but that's just
+    // less convenient.
+    // This is the only way to avoid copying the TestInfo stuff everywhere :/
+    // (I don't think micronaut can inject this object, unfortunately)
+    protected lateinit var testInfo: TestInfo
+    protected lateinit var testPrettyName: String
+
+    @BeforeEach
+    fun getTestInfo(testInfo: TestInfo) {
```

Comment:
can we have several beforeEach functions that will all get executed? If not, you'll have to set a closed `@BeforeEach`, and an `open executeBeforeEach`

Reply:
junit supports multiple BeforeXyz functions, but doesn't guarantee ordering amongst them. I think that's fine for now, we shouldn't ever need the test name during other test setup

(but yes, if that becomes a problem, what you're describing sounds right)